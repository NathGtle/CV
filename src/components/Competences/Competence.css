.Competence {
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  margin: var(--space-s) 0;
}

.nameSkill {
  grid-column: 1 / span 6;
  grid-row: 1;
}
