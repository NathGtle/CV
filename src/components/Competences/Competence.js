import React from 'react'
import Level from '../Level/Level'
import './Competence.scss'

const Competence = ({ competence }) => (
  <>
    <article className="Competence">
      <span className="nameSkill">{competence.name}</span>
      <Level level={competence.value} />
    </article>
  </>
)

export default Competence
