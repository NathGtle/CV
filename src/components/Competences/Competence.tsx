import React from 'react'
import PropTypes from 'prop-types'
import Level from '../Level/Level'
import './Competence.scss'

interface CompetenceInterface {
  competence: {
    name: string
    value: string
  }
}

const Competence = ({ competence }: CompetenceInterface) => {
  if (!competence) return null
  return (
    <article className="Competence">
      <span className="nameSkill">{competence.name}</span>
      <Level level={competence.value} />
    </article>
  )
}

Competence.propTypes = {
  competence: PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  }),
}

export default Competence
