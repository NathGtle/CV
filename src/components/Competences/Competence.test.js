import React from 'react'
import renderer from 'react-test-renderer'
import Competence from './Competence'

const mocks = {
  competence: {
    name: 'Javascript',
    value: '4.5',
  },
}

jest.mock('../Level/Level', () => 'mock-level')

describe('Competence component', () => {
  it('should render correctly with props', () => {
    const component = renderer.create(<Competence {...mocks} />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly without props', () => {
    const component = renderer.create(<Competence />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
