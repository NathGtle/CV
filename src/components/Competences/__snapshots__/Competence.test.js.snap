// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competence component should render correctly with props 1`] = `
<article
  className="Competence"
>
  <span
    className="nameSkill"
  >
    Javascript
  </span>
  <mock-level
    level="4.5"
  />
</article>
`;

exports[`Competence component should render correctly without props 1`] = `null`;
